### ISSUE TITLE
Negative Loss on gpu ALS model

### ISSUE THREAD

**Issue by sikhad** [benfred/implicit#367]
I'm getting a negative loss value when running ALS using GPU (loss = -.0346) regardless of varying all parameters. When running the same data/parameters on CPU, I'm getting a positive loss. I'm confused why loss could be negative. 

It's a ~6500 x 1m csr matrix.

``` python
params = {'factors':64, 
          'use_gpu':True, 
          'use_native':True, 
          'use_cg':True, 
          'regularization':0, 
          'num_threads':0,
          'iterations':5,
          'calculate_training_loss':True}

# initialize a model
model = implicit.als.AlternatingLeastSquares(**params)

# train the model on a sparse matrix of item/user/confidence weights
model.fit(csr, show_progress=True)
```

**Comment by benfred** [comment]
Its looking like the GPU loss calculation might be buggy (See also #441  )

**Comment by gallir** [comment]
Thanks. Will you release a new pip module version?

**Comment by benfred** [comment]
@gallir - I'm working on getting a new version together -  I also want to get changes like https://github.com/benfred/implicit/pull/661 and https://github.com/benfred/implicit/pull/656 pushed out to people too.

I'd also like to fix the conda packaging errors with this version - once I have a handle on that I'll push out a new release.

**Comment by benfred** [comment]
@gallir - fix is on v0.7.0
